c10/xpu/XPUCachingAllocator.h

```diff
@@ -4,7 +4,17 @@
 #include <c10/core/CachingDeviceAllocator.h>
 #include <c10/xpu/XPUStream.h>
 
+
 namespace c10::xpu::XPUCachingAllocator {
+class XPUAllocator : public DeviceAllocator {
+ public:
+  virtual void init(DeviceIndex device_count) = 0;
+  virtual void* raw_alloc(size_t size) = 0;
+  virtual void* raw_alloc_with_stream(size_t size, XPUStream stream) = 0;
+  virtual void raw_delete(void* ptr) = 0;
+  virtual getMemoryFraction(DeviceIndex device) = 0;
```

Comment:
We really need `getMemoryFraction` and `setMemoryFraction` here to support the Memory Pool? If no, let's remove it.

Reply:
> We really need `getMemoryFraction` and `setMemoryFraction` here to support the Memory Pool? If no, let's remove it.
I followed CUDA here, and I think expose Allocator is not for mempool